controllers/job.controller.js

```diff
@@ -0,0 +1,52 @@
+const User = require('../models/user.model')
+const Job = require('../models/job.model')
+const Company = require('../models/company.model')
+const sendNotificationReq = require('../utils/sendEmailRequest')
+
+exports.createJob = async (req,res)=>{
+    try{
+        const jobData = {
+            title : req.body.title,
+            description : req.body.description,
+            vacancies : req.body.vacancies,
+            postedBy : req.user._id,
+            company : req.user.companyId
+        }
+    
+        const jobCreated = await Job.create(jobData);
+        if(jobCreated){
+            const poster = req.user;  // got user from JWT userID in middlewere
+
+            poster.activeJobsPosted.push(jobCreated._id);
+            await poster.save();
+
+            const company = await Company.findOne({_id : req.user.companyId});
```

Comment:
What if the company is not present.

Add the validation for it in the middlewares